### ISSUE TITLE
RANGE_ARRAY_OFFSET not detected when array accessed with negative index

### ISSUE THREAD

**Issue by csrma** [spotbugs/spotbugs#3462]
In the following code, `if (offset < 0)` check makes sure that the `offset` value is negative, and then still accesses the `numbers` array, which will certainly result in ArrayIndexOutOfBoundsException. However, SpotBugs does not detect this issue in this code.
```
class ArrayOffsetBug {

    public int showBug(int[] numbers, int offset) {
        if (offset < 0)
            return numbers[offset]; // BUG: RANGE: Array offset is out of bounds (RANGE_ARRAY_OFFSET)
        return numbers[offset];
    }

    public static void main(String[] args) {
        ArrayOffsetBug bugDemo = new ArrayOffsetBug();
        int[] sampleArray = {1, 2, 3, 4, 5};
        int result = bugDemo.showBug(sampleArray, -2);
        System.out.println(result);
    }
}
```
SpotBugs version: `4.9.2`
Java: `Temurin openjdk 21.0.7`
Ran from CLI as the following two commands:
```
export SPOTBUGS_JAR=/path/to/spotbugs.jar
```
```
javac ArrayOffsetBug.java && \
java -jar $SPOTBUGS_JAR \
    -low \
    -effort:max \
    ArrayOffsetBug.class
```

**Comment by gtoison** [comment]
Hello,
This bug code is for methods such as System.arraycopy() not for the index of an array. 
There's apparently no detector checking that the index is within the expected range of an array.
